src/compiler/compile/nodes/Element.ts

```diff
@@ -395,26 +395,20 @@ export default class Element extends Node {
 					component.slot_outlets.add(name);
 				}
 
-				let ancestor = this.parent;
-				do {
-					if (ancestor.type === 'InlineComponent') break;
-					if (ancestor.type === 'Element' && /-/.test(ancestor.name)) break;
-
-					if (ancestor.type === 'IfBlock' || ancestor.type === 'EachBlock') {
-						const type = ancestor.type === 'IfBlock' ? 'if' : 'each';
-						const message = `Cannot place slotted elements inside an ${type}-block`;
+				if (parent.type === 'IfBlock' || parent.type === 'EachBlock') {
```

Comment:
I'm leaning towards removing this special check for this error, and just handling it below. What do you think?

Reply:
Sounds good.